test/ng/directive/inputSpec.js

```diff
@@ -840,6 +840,16 @@ describe('input', function() {
 
         expect($rootScope.form.alias.$error.max).toBeFalsy();
       });
+
+      it('should validate when timezone is provided.', function() {
+        inputElm = helper.compileInput('<input type="date" ng-model="value" name="alias" ' +
+            'min="2013-01-01" max="2013-12-31" ng-model-options="{timezone: \'UTC\', allowInvalid: true}" required/>');
+        $rootScope.value = new Date(Date.UTC(2013, 11, 31));
+        $rootScope.$digest();
+
+        expect($rootScope.form.alias.$error.max).toBeFalsy();
+        expect($rootScope.form.alias.$valid).toBeTruthy();
```

Comment:
There should be am expectation after this that checks whether the validation also works when the input is updated. You might have to clear the model for that first.